js/gloss-witness.js

```diff
@@ -611,10 +612,27 @@ function initFragmentForm(event){
         getAllWitnessFragmentsOfSource(selections)
     }
     else{
-        addEventListener('source-text-loaded', ev => {
-           getAllWitnessFragmentsOfSource(selections)
-        })
-        if(!textSelectionElem.getAttribute("source-uri")) textSelectionElem.setAttribute("source-uri", sourceURI)
+        if(source.startsWith("http")) {
+            sourceURI = source
```

Comment:
it might be more reliable to URL test this, but I think this works for our purposes.